### PR TITLE
obs-stream-effects: Add 3D Transform, Blur and Displacement Map

### PR DESCRIPTION
**Pull request by Xaymar**
The initial version adds '3D Transform', 'Displacement Map' and 'Blur' filters, which can be used to create more advanced scenes.

1. '3D Transform' allows for semi-3D (non-interacting with other scene items) effects. It can be used to place a source into a 3D rendered image, for example to place a browser source chat into a rendered phone on a desk.
2. 'Displacement Map' is simply a displacement map.
3. 'Blur' adds Box, Gaussian, and Bilateral blur. Each of these is kind of GPU hungry so they should be used sparingly, if at all.

See #960 for the old PR containing screenshots.

### CODE REVIEW

File: plugins/obs-stream-effects/.gitignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## Ignore file for OBS-AMD-Media-Framework.
```
Comment (RytoEX) [inline plugins/obs-stream-effects/.gitignore]: I think you have a leftover line from your AMF plugin.

File: plugins/obs-stream-effects/source/version.h.in [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/*
+ * A Plugin that integrates the AMD AMF encoder into OBS Studio
```
Comment (RytoEX) [inline plugins/obs-stream-effects/source/version.h.in]: I think you have a leftover line from your AMF plugin.

File: plugins/obs-stream-effects/source/version.h.in [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * Copyright (C) 2016 - 2017 Michael Fabian Dirks
```
Comment (RytoEX) [inline plugins/obs-stream-effects/source/version.h.in]: Copyright years are generally meant to cover years of public releases of software, so this should probably be just 2017 like the rest of the files.

**Comment by Xaymar** [comment]
Thanks @RytoEX, I've fixed them now.

**Comment by Obyoxar** [comment]
I find it kinda funny, that both of us programmed a 3D-Transform-filter at the same time. I'll definitely check out yours, maybe you do better than me :P.

**Comment by Xaymar** [comment]
@Obyoxar This one is actually over a month old. I was waiting for Jim to fork it, but it seems that that takes a while - this one is just a pure code copy from the original repo.

**Comment by Obyoxar** [comment]
@Xaymar Ouo, really? Well apparently I didn't look close enough when I was searching for this kind of plugin, but hey, I guess two are better than one, hehe.

**Comment by Xaymar** [comment]
@Obyoxar See PR #960

**Comment by Obyoxar** [comment]
@Xaymar Good work indeed, I'd really like to see this filter in the source code one time, so pls carry on.

**Comment by ONSBalder** [comment]
Can one of the admins verify this patch?

**Comment by Xaymar** [comment]
This has been moved back into its own plugin repository here: https://github.com/Xaymar/obs-stream-effects

An official build will be released with 0.3.0, the first "stable" version.

**Pull request closed without merging**